File: sdk/src/main/java/com/robotemi/sdk/listeners/OnConstraintBeWithStatusChangedListener.java
package com.robotemi.sdk.listeners;

import androidx.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public interface OnConstraintBeWithStatusChangedListener {

    /**
     * Listen for Constraint Follow state.
     *
     * @param isConstraint true if temi is in constraint follow state, false if is in any other state.
     */
    void onConstraintBeWithStatusChanged(boolean isConstraint);
}

File: sdk/src/main/java/com/robotemi/sdk/listeners/OnConstraintBeWithStatusChangedListener.kt
package com.robotemi.sdk.listeners

interface OnConstraintBeWithStatusChangedListener {

    /**
     * Listen for Constraint Follow state.
     *
     * @param isConstraint true if temi is in constraint follow state, false if is in any other state.
     */
    fun onConstraintBeWithStatusChanged(isConstraint: Boolean)
}
